pubsub/src/main/java/com/github/twitch4j/pubsub/TwitchPubSub.java

```diff
@@ -564,6 +564,16 @@ public void onTextMessage(WebSocket ws, String text) {
                                         log.warn("Unparsable Message: " + message.getType() + "|" + message.getData());
                                         break;
                                 }
+                            } else if (topic.startsWith("user-moderation-notifications")) {
+                                int lastPeriod = topic.lastIndexOf('.');
+                                String userId = topic.substring(topic.indexOf('.') + 1, lastPeriod);
```

Comment:
Maybe we can improve this a little by adding sth. after https://github.com/twitch4j/twitch4j/blob/39df41f85ce6cdc6810c4cdeb04a347aa0bbb1e4/pubsub/src/main/java/com/github/twitch4j/pubsub/TwitchPubSub.java#L369 to parse the topic value first (split on `.`).

And then check for a exact match on the topic name and check that the param count (ie. 3 for this one, topic + userId + channelId) matches.
We wouldn't need the lastIndexOf / substring code in every case then.

Reply:
Good idea, I switched to the split approach for this topic, will migrate the others in a separate refactor PR